WordPress/src/main/java/org/wordpress/android/ui/posts/adapters/PostsListAdapter.java

```diff
@@ -191,10 +191,20 @@ private boolean canShowStatsForPost(PostModel post) {
     }
 
     private boolean canPublishPost(PostModel post) {
-        return post != null && !UploadService.isPostUploadingOrQueued(post) &&
+        // TODO remove the hasFailedMedia(post) check when we get a proper retry mechanism in place,
+        // that retried to upload any failed media along with the post
+        return post != null && !UploadService.isPostUploadingOrQueued(post) && !hasFailedMedia(post) &&
                 (post.isLocallyChanged() || post.isLocalDraft() || PostStatus.fromPost(post) == PostStatus.DRAFT);
     }
 
+    private boolean hasFailedMedia(PostModel post) {
```

Comment:
This method doesn't really feel like it belongs in the `PostsListAdapter` - what do you think about moving it to the `UploadService` and calling it `hasMediaErrorForPost()`?